src/server/handlers/thirdpartyRequests/transactions/{ID}/authorizations.ts

```diff
@@ -0,0 +1,164 @@
+ Unless required by applicable law or agreed to in writing, the Mojaloop
+ files are distributed onan 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Raman Mangla <ramanmangla@google.com>
+ --------------
+ ******/
+
+import Logger from '@mojaloop/central-services-logger'
+import { Enum } from '@mojaloop/central-services-shared'
+import { Consent } from '../../../../../model/consent'
+import { Scope } from '../../../../../model/scope'
+import { consentDB, scopeDB } from '../../../../../lib/db'
+import { NotFoundError } from '../../../../../model/errors'
+import { verifySignature } from '../../../../../lib/challenge'
+import { thirdPartyRequest } from '../../../../../lib/requests'
+import { Request, ResponseToolkit, ResponseObject } from '@hapi/hapi'
+import {
+  AuthPayload,
+  isPayloadPending,
+  hasActiveCredentialForPayload,
+  hasMatchingScopeForPayload,
+  putErrorRequest
+} from '../../../../../domain/authorizations'
+
+/*
+ * Asynchronous POST handler helper function to
+ * process everything in the background
+ */
+export async function validateAndVerifySignature (
+  request: Request): Promise<void> {
+  const payload: AuthPayload = request.payload as AuthPayload
+
+  // Validate incoming payload status
+  if (!isPayloadPending(payload)) {
+    return putErrorRequest(request, '3100', 'Bad Request')
+  }
+
+  let consent: Consent
+
+  // Check if consent exists and retrieve consent data
+  try {
+    consent = await consentDB.retrieve(payload.consentId)
+  } catch (error) {
+    Logger.push(error)
+    Logger.error('Could not retrieve consent')
+
+    if (error instanceof NotFoundError) {
+      return putErrorRequest(request, '2000', 'Not Found')
+    }
+
+    return putErrorRequest(request, '2000', 'Server Error')
+  }
+
+  let consentScopes: Scope[]
+
+  // Retrieve scopes for the consent
+  try {
+    consentScopes = await scopeDB.retrieveAll(payload.consentId)
+  } catch (error) {
+    Logger.push(error)
+    Logger.error('Could not retrieve scope')
+
+    if (error instanceof NotFoundError) {
+      return putErrorRequest(request, '2000', 'Forbidden')
+    }
+
+    return putErrorRequest(request, '2000', 'Server Error')
```

Comment:
Maybe this should be `2001`

https://docs.mojaloop.io/mojaloop-specification/documents/API%20Definition%20v1.0.html#76-error-codes